csvcubed/csvcubed/readers/columnschema.py

```diff
@@ -0,0 +1,447 @@
+                    units.append(ExistingQbUnit(unit_uri=self.values.from_existing))
+
+                else:
+                    raise ValueError(f"Unexpected unit value: {unit}")
+
+            return QbMultiUnits(units)
+
+        raise ValueError(f"Unhandled 'Units' value: {self}")
+
+
+@dataclass
+class MeasureType(SchemaBaseClass):
+    label: Optional[str] = None
+    description: Optional[str] = None
+    from_existing: Optional[str] = None
+    definition_uri: Optional[str] = None
+
+
+@dataclass
+class Measures(SchemaBaseClass):
+    values: Union[bool, List[MeasureType]] = True
+
+    def map_to_multi_measure_dimension(
+        self, data: PandasDataTypes
+    ) -> QbMultiMeasureDimension:
+        # When values is a single bool True then create new Measures from the csv column data
+        if self.values is True:
+            return QbMultiMeasureDimension.new_measures_from_data(data)
+
+        elif isinstance(self.values, list):
+            new_measures = []
+            for new_measure in self.values:
+                if not isinstance(new_measure, MeasureType):
+                    raise ValueError(f"Unexpected measure: {new_measure}")
+
+                if any([new_measure.label, new_measure.description, new_measure.definition_uri]):
+                    # NewQBMeasure
+                    new_measures.append(_map_measure(new_measure))
+                else:
+                    new_measures.append(QbMultiMeasureDimension(new_measure))
+
+            return QbMultiMeasureDimension(new_measures)
+
+        else:
+            raise ValueError(f"Unexpected values type: {self.values}")
+
+
+@dataclass
+class ObservationValue(SchemaBaseClass):
+    datatype: Optional[str] = None
+    unit: Union[None, str, Units] = None
+    measure: Union[None, str, Measures] = None
+
+    def map_to_qb_observation(self) -> QbObservationValue:
+        unit = None
+        if isinstance(self.unit, str):
+            unit = ExistingQbUnit(unit_uri=self.unit)
+
+        elif isinstance(self.unit, Units):
+            unit = _map_unit(self.unit)
+
+        elif self.unit is not None:
+            raise ValueError(f"Unexpected unit: {self.unit}")
+
+        if self.measure is None:
+            # Multi-measure cube
+            return QbMultiMeasureObservationValue(
+                data_type=self.datatype or "decimal", unit=unit
+            )
+        else:
+            # Single measure qb
+            measure = None
+            if isinstance(self.measure, str):
+                measure = ExistingQbMeasure(self.measure)
+
+            elif isinstance(self.measure, Measures):
+                measure = _map_measure(self.measure)
+
+            else:
+                raise ValueError(f"Unhandled measure type: {self.measure}")
+
+            return QbSingleMeasureObservationValue(
+                measure=measure, unit=unit, data_type=self.datatype or "decimal"
+            )
+
+
+def _map_unit(resource: Units) -> NewQbUnit:
```

Comment:
I don't think the input type is correct here. It should be `resource: UnitType` given your current naming.